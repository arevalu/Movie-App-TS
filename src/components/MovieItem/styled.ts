import styled from 'styled-components';
import { theme } from '../../styles/theme/theme';
import { mq } from '../../styles/breakpoints';

const { colors, borderRadius, font, spacing } = theme;

type CardProps = {
    imgSource: string;
}

export const Card = styled.div`
    cursor: pointer;
    margin-bottom: ${spacing.large};
    position: relative;
    width: 80%;

    ${mq[1]} {
        width: 40%;
    }

    ${mq[2]} {
        width: 30%;
    }

    ${mq[3]} {
        width: 200px;
    }
`

export const VoteAverage = styled.div`
    background-color: ${colors.secondary};
    border-radius: ${borderRadius.sm};
    color: ${colors.dark};
    display: flex;
    align-items: center;
    justify-content: center;
    height: 25px;
    position: absolute;
    top: 15px;
    right: 5%;
    width: 50px;

    span {
        font-size: ${font.size.small};
        font-weight: 600;
    }
`

export const CardMedia = styled.div<CardProps>`
    background-image: url( "${ ({imgSource}) => imgSource}" );
    background-repeat: no-repeat;
    background-size: cover;
    background-position: center;
    border-radius: ${borderRadius.lg};
    padding-bottom: 150%;
    width: 100%;

    ${mq[3]} {
        border-radius: ${borderRadius.md};
    }
`

export const CardTitle = styled.div`
    margin-top: 0.5rem;

    p {
        font-size: ${font.size.h4};
        font-weight: 600;
        padding-left: 10px;

        ${mq[3]} {
            font-size: ${font.size.base};
        }   
    }
`