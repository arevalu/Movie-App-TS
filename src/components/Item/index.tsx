import React from 'react';

import { URL_IMAGE_POSTER } from '../../utils/const';
import { popularMovie } from '../../types/movies';
 
import { Card, CardMedia, CardTitle, VoteAverage } from './styled';

type ItemProps = {
    movie: popularMovie;
}

const MovieItem: React.FC<ItemProps> = ({ movie }) => {

    const { poster_path, title, vote_average } = movie;

    return (
        <Card>
            <VoteAverage>
                <span>{ vote_average }</span>
            </VoteAverage>
            <CardMedia
                role="img"
                imgSource={ `${ URL_IMAGE_POSTER }${ poster_path }` }
            />
            <CardTitle>
                <p>{ title }</p>
            </CardTitle>
        </Card>
    )
}

export default MovieItem;
