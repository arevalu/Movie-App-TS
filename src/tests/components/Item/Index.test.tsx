import React from 'react';
import { render } from '@testing-library/react';

import MovieItem from '../../../components/Item/index';
import { CardMedia } from '../../../components/Item/styled';

describe('Pruebas en componente <Item />', () => {

    const movie = {
        backdrop_path: "/srYya1ZlI97Au4jUYAktDe3avyA.jpg",
        genre_ids: ["Fantasy","Action","Adventure"], 
        id: 464052,
        overview: "Wonder Woman comes into conflict with the Soviet Union during the Cold War in the 1980s and finds a formidable foe by the name of the Cheetah.",
        poster_path: "/8UlWHLMpgZm9bx6QYh0NFoq67TZ.jpg",
        title: "Wonder Woman 1984",
        vote_average: 7.2
    }
    
    test('<Item /> debe renderizar correctamente', () => {
        const { container } = render( <MovieItem movie={ movie } /> );

        expect(container).toMatchSnapshot();
    });

    test('Title debe renderizar en tag <p>', () => {
        const { getByText } = render( <MovieItem movie={ movie } /> );

        const movieTitle = getByText(movie.title);

        expect( movieTitle ).toBeDefined();
        expect( movieTitle.tagName.toLowerCase() ).toBe('p');
    });
    
    test('Vote_average debe renderizar en tag <span>', () => {
        const { getByText } = render( <MovieItem movie={ movie } /> );

        const voteAverage = getByText(movie.vote_average);

        expect( voteAverage ).toBeDefined();
        expect( voteAverage.tagName.toLowerCase() ).toBe('span');
    });
    
    test('La prop imgSource de CardMedia debe ser imageUrlBase + poster_path', () => {
        const { getByRole } = render( <MovieItem movie={ movie } /> );

        const imageUrlBase = new RegExp('https://image.tmdb.org/t/p/original');
        const CardMedia_styled = getByRole( 'img' );

        expect( CardMedia_styled ).toHaveStyleRule('background-image', imageUrlBase);
    });

});