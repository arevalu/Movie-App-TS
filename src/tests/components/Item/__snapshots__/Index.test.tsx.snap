// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <Item /> <Item /> debe renderizar correctamente 1`] = `
.c0 {
  cursor: pointer;
  margin-bottom: 3.75rem;
  position: relative;
  width: 80%;
}

.c1 {
  background-color: #ffc107;
  border-radius: 0.25rem;
  color: #181C1F;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 25px;
  position: absolute;
  top: 15px;
  right: 5%;
  width: 50px;
}

.c2 {
  background-image: url( "https://image.tmdb.org/t/p/original/8UlWHLMpgZm9bx6QYh0NFoq67TZ.jpg" );
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  border-radius: 1.125rem;
  padding-bottom: 150%;
  width: 100%;
}

.c3 {
  margin-top: 1.25rem;
}

.c3 p {
  font-size: 1.25rem;
  font-weight: 600;
  padding-left: 10px;
}

@media (min-width:480px) {
  .c0 {
    width: 40%;
  }
}

@media (min-width:768px) {
  .c0 {
    width: 30%;
  }
}

@media (min-width:960px) {
  .c0 {
    width: 200px;
  }
}

@media (min-width:960px) {
  .c2 {
    border-radius: 0.75rem;
  }
}

@media (min-width:960px) {
  .c3 p {
    font-size: 1rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <span>
        7.2
      </span>
    </div>
    <div
      class="c2"
      role="img"
    />
    <div
      class="c3"
    >
      <p>
        Wonder Woman 1984
      </p>
    </div>
  </div>
</div>
`;
